Puara/PeakDetection.hpp

```diff
@@ -2,44 +2,130 @@
 
 #include "3rdparty/extras/PeakDetector.h"
 
+#include <array>
 #include <halp/audio.hpp>
 #include <halp/controls.hpp>
 #include <halp/meta.hpp>
+#include <halp/controls.enums.hpp>
 #include <puara/gestures.h>
 
-#include <halp/controls.enums.hpp>
+// data_port struct to include descriptions
+namespace halp {
+template <static_string lit, static_string desc, typename T>
+struct data_port : val_port<lit, T>
+{
+  using base_t = val_port<lit, T>;
+  using base_t::operator=;   
+
+  static clang_buggy_consteval auto description() { return std::string_view{desc.value}; }
+};
+} // namespace halp
 
 namespace puara_gestures::objects
 {
+
+// parameter watcher to trigger when a parameter is changed
+template<typename T> 
+struct ParameterWatcher { 
```

Comment:
Same, this looks like some generic appliance that  can be used in multiple objects